### ISSUE TITLE
Trino with iceberg catalog using glue: not returning data

### ISSUE THREAD

**Issue by nikhilm29** [trinodb/trino#22779]
Environment Details:
Trino Version: 433
Iceberg with glue 

Issue Description:
After successfully inserting 916 records into the database using Trino version 433, we confirmed the presence of the corresponding Parquet file. However, when executing a SELECT statement to retrieve these records, the result set appears empty

Summary of Snapshot:
`{changed-partition-count=1, added-data-files=1, total-equality-deletes=0, added-records=916, trino_query_id=20240719_102256_00356_rgjsk, total-position-deletes=0, added-files-size=63836, total-delete-files=0, total-files-size=47805891, total-records=582454, total-data-files=299}`
![image](https://github.com/user-attachments/assets/5572deef-cec2-478f-9573-b763dfd8775c)

Additional Details:
Trino Query ID: 20240719_102256_00356_rgjsk
Inserted Data: 916 records

Issue Observation:
Despite the successful insertion and the presence of the Parquet file, querying the data returns no results.

Expected Behavior:
The SELECT statement should return the 916 records that were successfully inserted.

Actual Behavior:
The SELECT statement returns an empty result set.

**Comment by ebyhr** [comment]
I don't think we can help with the provided information. Please feel free to reopen with additional info if you still face the issue with the latest version.
